Bootstrap 4

ohmbt opened the pull request.

ohmbt committed:
Initial Bootstrap 4 changes

Updated local copy of Bootstrap to v4.3.1
Updated Datepicker for Bootstrap to v1.9.0 for Bootstrap 4 support
Replaced instances of 'form-control-static' with
'form-control-plaintext'
Replaced instances of 'help-block' with 'form-text text-muted'

ohmbt committed:
Added changelog entry for 2.0.0

Updated errorlists to include Bootstrap 4's `.invalid-feedback` class
Added `widgets.CheckboxInput` which includes the proper Bootstrap 4
class
Updated `bootstrap/field.html` and `bootstrap/field_readonly.html` to
respect new Bootstrap 4 structure for checkboxes and radios

ohmbt committed:
Added div with `.controls.clearfix` back in so it can be used again

Moved the above div's closing tag to after the errorlist so
`.invalid-feedback` will display properly

ohmbt committed:
Updated changelog

Updated a link to Bootstrap's documentation
Restored `.clearfix` to the rendered_value div in field_readonly.html

dcwatson reviewed (changes requested):
This generally looks pretty good. Two things:

1. It's minor, but maybe change the non_field_errors alerts in form.html to use markup like https://getbootstrap.com/docs/4.3/components/alerts/#dismissing - i.e. &times; instead of an icon font, at least.

2. Doesn't look like the pager.html template was updated, but it looks like BS4 added some classes on the li/a elements inside of the pager - https://getbootstrap.com/docs/4.3/components/pagination/

ohmbt committed:
Replaced "fa fa-times" with "&times;"

Made the alert fade out when dismissed, as recommended by Bootstrap.
Rearranged the alert text and button in the code so it reflects how they
actually display (text first, then the button).
Added "page-item" and "page-link" classes where appropriate in
pagination.
Moved help text and invalid-feedback to be siblings of the last item for
RadioSelects and CheckboxSelectMultiples, so invalid-feedback displays
properly when there are errors, and so their field ordering is
consistent with the other widgets.
Gave Select, SelectMultiple, and NullBooleanSelect widgets the
"custom-select" class so they style more cleanly when invalid.

ohmbt committed:
Changelog update
